mgr/TTLCache: fix PyObject* lifetime management and cleanup logic (#65245)
ceph/ceph: doc/mgr/prometheus.rst

    ceph_health_detail{name="OSD_DOWN",severity="HEALTH_WARN"} 1.0
    ceph_health_detail{name="PG_DEGRADED",severity="HEALTH_WARN"} 1.0

> The module also maintains an in-memory history of health check stats

Reviewer: Missing a full stop here. This will render in the same paragraph as the following sentence so it will look like they are a single sentence. Would "statistics" be better than "stats"?

```suggestion
The module also maintains an in-memory history of health check statistics.
```

> By default, it retains a maximum of 1000 entries, These limits are configurable via the following runtime option:

Reviewer: Something strange going on with the commas / capitalization and I suggest to rewrite the beginning of the first sentence (but I might be wrong). And because now there is only 1 option instead of the 2 before, talk about a singular "limit":

```suggestion
By default the history retains a maximum of 1000 entries. This limit is configurable via the following runtime option:
```

>   ``mgr/prometheus/healthcheck_history_max_entries`` - the maximum number of unique health check to track in memory (default: 1000).

Reviewer: Should have plural here I think so I added "entries" that's consistent with the previous text.

```suggestion
  ``mgr/prometheus/healthcheck_history_max_entries`` - the maximum number of unique health check entries to track in memory (default: 1000).
```

> These settings help avoid unbounded memory growth in large or long-lived clusters.

Reviewer: Same singular "setting"

```suggestion
This setting helps avoid unbounded memory growth in large or long-lived clusters.
```
